osu.Game/Overlays/Settings/Sections/Graphics/RendererSettings.cs

```diff
@@ -67,11 +67,7 @@ private void load(FrameworkConfigManager config, OsuConfigManager osuConfig, IDi
                 if (r.NewValue == RendererType.Automatic && automaticRendererInUse)
                     return;
 
-                if (game?.RestartAppWhenExited() == true)
-                {
-                    game.AttemptExit();
-                }
-                else
```

Comment:
The flow here makes me wonder whether `Restart` should handle the confirmation for platforms which can't restart themselves, rather than having this handled on each usage.

But then again, the usages may be so few and far between that it's fine to leave it as-is?

Reply:
I'd say per-usage is better because places like tournament client use `Restart()` and I don't really wanna think about how to make that work well across such varied usages.